lsusb.c

```diff
@@ -3829,6 +3937,13 @@ int main(int argc, char *argv[])
 			"  -d vendor:[product]\n"
 			"      Show only devices with the specified vendor and\n"
 			"      product ID numbers (in hexadecimal)\n"
+			"  -f FORMAT\n"
```

Comment:
You have two different things being added in one change.  One is the format operation, which is nice to have.  The second is your new ability to handle serial numbers in hex format, which is different.

Please break this up into at least 2 different changes and resubmit them as a patch series.

Reply:
Yes, right. I thought better to add some also useful functions instead of making a quick and dirty modification.
You really need to split? I hoped to show you that It's a useful feature even if normally only needed with devices, which breaks the rules of the serial number descriptor.
To be honest: If you make a release without the %x output it's useless for me :-(
It was the main reason why I made this changes. 
And there are definitely devices with this bug :-(
If you really need it in two steps, I can split. Please send a note.